src/components/recovery/Recovery.jsx

```diff
@@ -0,0 +1,260 @@
+import "./Recovery.scss";
+import nextPage from "./nextPage"
+import formValidation from "./formValidation"
+import validate from "./validate"
+
+let interval;
+const Recovery = (props) => {
+  const location = useLocation();
+  const [email, setEmail] = useState("");
+  const [number, setNumber] = useState("");
+  const [emailMessage, setEmailMessage] = useState("");
+  const [numberMessage, setNumberMessage] = useState("");
+  const userObj = props.userReducer;
+  const [method, setMethod] = useState(userObj.method);
+  const [seconds, setSeconds] = useState(0);
+  const [min, setMin] = useState(0);
+  const [isTimer, setIsTimer] = useState(false);
+  const history = useHistory();
+  let reduxMin = userObj.min;
+  let reduxSeconds = userObj.seconds;
+
+  useEffect(() => {
+    const lang = props.languageReducer.lang;
+    Localization.setLanguage(lang);
+  }, [props.languageReducer.lang]);
+
+  const checkResend = ()=>{
+    const resendStorage = getResend();
+    if (reduxMin >= 0) {
+      setMin(reduxMin);
+    }
+
+    if (reduxSeconds >= 0) {
+      setSeconds(reduxSeconds);
+    }
+    if (location.state && reduxMin === 0 && reduxSeconds === 0) {
+      let locationMin = location.state.min;
+      let locationSeconds = location.state.seconds;
+      if (locationSeconds > 0) {
+        setSeconds(locationSeconds);
+      }
+      if (locationMin !== 0) {
+        setMin(locationMin);
+      }
+    } else {
+      if (resendStorage === "third" && reduxMin === 0 && reduxSeconds === 0) {
+        setMin(15);
+        setSeconds(0);
+      }
+    }
+
+    interval = setInterval(() => {
+      setIsTimer(true);
+      setSeconds((seconds) => seconds - 1);
+    }, 1000);
+  }
+  
+  useEffect(() => {
+    checkResend()
+  }, []);
+
+  
+
+  useEffect(() => {
+    const changeTimer =()=> {
+      if (seconds < 0 && min > 0) {
+        setMin((min) => min - 1);
+        setSeconds(59);
+      }
+      if (seconds === 0 && min === 0 && isTimer) {
+        clearInterval(interval);
+      }
+    }
+    
+    changeTimer()
+  }, [seconds, min]);
+
+  useEffect(() => {
+    userObj.min = min;
+  }, [min]);
+
+  useEffect(() => {
+    if (seconds >= 0) {
+      userObj.seconds = seconds;
+    }
+  }, [seconds]);
+
+
+  const validateHandler = () => {
+    const isValidEmail = validate(formValidation.email, email ,
+        setEmailMessage,
+        setNumberMessage  );
+    const isValidNumber = validate(formValidation.number, number ,
+        setEmailMessage,
+        setNumberMessage);
+
+    return isValidEmail || isValidNumber;
+  };
+
+  const handleSetMethod = (type) => {
+    setNumberMessage("");
+    setEmailMessage("");
+    setMethod(type);
+  };
+
+
+
+  return (
+    <>
+      <div className="form_container recovery_container">
+        <div className="form_wrapper">
+          <h1 className="form_title mb-8">
+            {Localization.title}, {props.userReducer.firstname || "client"}!{" "}
```

Comment:
Remove !{" "}, if you need a space add a margin right

Reply:
That's add by default from prettier sometimes I will remove it